opal/util/output.c

```diff
@@ -176,8 +176,7 @@ bool opal_output_init(void)
     } else {
         verbose.lds_want_stderr = true;
     }
-    gethostname(hostname, sizeof(hostname));
-    hostname[sizeof(hostname)-1] = '\0';
+    gethostname(hostname, MAXHOSTNAMELEN);
```

Comment:
From the Linux `gethostname(2)` man page:

> SUSv2 guarantees that "Host names are limited to 255 bytes".  POSIX.1 guarantees that "Host names (not including the terminating null  byte) are  limited  to HOST_NAME_MAX bytes".  On Linux, HOST_NAME_MAX is defined with the value 64, which has been the limit since Linux 1.0 (earlier kernels imposed a limit of 8 bytes).

For POSIX systems, I think defining a MAXHOSTNAMELEN = limits.h:HOST_NAME_MAX + 1 and using it in this fashion is fine. We have space for the null terminator which gethostname() will append provided it doesn't overrun. I imagine your concern is for non-POSIX systems (which I also feared a bit).

Perhaps the alternative would be to do:

    +/* From limits.h. Since we're using gethostname(), we add 1 to the max to allow
    + * for null termination.
    + */
    +#ifdef HOST_NAME_MAX
    +#define MAXHOSTNAMELEN HOST_NAME_MAX + 1
    +#else
    +#define MAXHOSTNAMELEN 255 + 1
    +#endif
    #endif

Or just set it to 255 + 1 across the board and not worry about pulling it via limits.h? BTW - what does OS X limits.h:HOST_NAME_MAX say?